### PR TITLE
add binary number parsing

### PR DESCRIPTION
**Pull request by khris190**
that's pretty much it, i needed(wanted) binary numbers, its my first contribution to any project ever so i hope its not stupid

### CODE REVIEW

File: include/argparse/argparse.hpp [hunk 1]
```diff
@@ -180,9 +181,10 @@ constexpr bool starts_with(std::basic_string_view<CharT, Traits> prefix,
 }
 
 enum class chars_format {
-  scientific = 0x1,
-  fixed = 0x2,
-  hex = 0x4,
+  scientific = 0xf1,
+  fixed = 0xf2,
+  hex = 0xf4,
+  bin = 0xf8,
```
Comment (shaiku) [inline include/argparse/argparse.hpp:187]: Why did you change the one-hot encoding to set all the upper bits?

Comment (khris190) [reply include/argparse/argparse.hpp:187]: because radix_2 = 2 collides with fixed in template <class T> struct parse_number<T, chars_format::fixed>

**Comment by ismagilli** [comment]
Hello! Congratulations on the first contribution. In addition to the code that adds new functionality, it is highly desirable to also write tests and a documentation with examples in `README.md`. As an example for tests, look at the `test/test_scan.cpp` file. And as for the documentation, if I'm not mistaken, there is only one place about the various integer formats -- the table before the "Default Arguments" section.

**Comment by p-ranav** [comment]
Merged in #306
